sphinx/builders/singlehtml.py

```diff
@@ -88,29 +88,47 @@ def _get_local_toctree(
         )
         return self.render_partial(toctree)['fragment']
 
+    def prefix_ids_with_docname(self, tree: nodes.document) -> None:
+        # Append docname to refids and ids using format document-<docname>#<id>.
+        # Compensates for loss of the pathname section of the href, that
+        # ensures uniqueness in the html builder.
```

Comment:
What do you think about adjusting the name of this method to `ensure_fully_qualified_refids`, or similar?

Note: when I suggest the terminology `fully-qualified` there, I'm borrowing it as a name from DNS: https://en.wikipedia.org/wiki/Fully_qualified_domain_name